File: ScriptingInJava/src/com/jdojo/script/ArrayConversion.java
// ArrayConversion.java
package com.jdojo.script;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class ArrayConversion {
	public static void printInts(int[] ids) {
		System.out.print("Inside printInts():");
		System.out.println(Arrays.toString(ids));
	}
	
	public static void printDoubles(double[] salaries) {
		System.out.print("Inside printDoubles(double[]):");
		System.out.println(Arrays.toString(salaries));
	}
	
	public static void printList(List<Integer> idsList) {
		System.out.print("Inside printList():");
		System.out.println(idsList);
	}
	
	public static void printMap(Map<?,?> phoneMap) {
		System.out.println("Inside printMap():" );
		phoneMap.forEach((key, value) -> {
			System.out.println("key = " + key + ", value = " + value);
		});
	}
}

File: ScriptingInJava/arrayconversion.js
// arrayconversion.js

var ArrayConversion = Java.type("com.jdojo.script.ArrayConversion");

ArrayConversion.printInts([1, 2, 3]);
ArrayConversion.printInts([1, "Hello", 3]); // "hello" is converted to 0

// Non-integers will be converted to correposning integers per Nashorn rules
// when Nashorn array is converted to a Java array. true and false are converted 
// to 1 and 0, and 10.3 is converted to 10.
ArrayConversion.printInts([1, true, false, 10.3]);

// Nashorn array to Java double[] conversion
ArrayConversion.printDoubles([10.89, "Hello", 3]);

// Nashorn array to Java List conversion
ArrayConversion.printList([10.89, "Hello", 3]);

// Nashorn array to Java Map conversion
ArrayConversion.printMap([10.89, "Hello", 3]);
